modules/phase_field/src/kernels/SingleGrainRigidBodyMotion.C

```diff
@@ -0,0 +1,62 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "SingleGrainRigidBodyMotion.h"
+
+template<>
+InputParameters validParams<SingleGrainRigidBodyMotion>()
+{
+  InputParameters params = validParams<Kernel>();
+  params.addClassDescription("Adds rigid mody motion to a single grain");
+  params.addRequiredCoupledVar("c", "Concentration");
+  params.addParam<unsigned int>("op_index",0, "Grain number for the kernel to be applied");
+  params.addParam<MaterialPropertyName>("advection_velocity", "Material property giving the advection velocity of grains");
+  params.addParam<MaterialPropertyName>("advection_velocity_divergence", "Material property for divergence of advection velocities");
+  return params;
+}
+
+SingleGrainRigidBodyMotion::SingleGrainRigidBodyMotion(const InputParameters & parameters) :
```

Comment:
```SingleGrainRigidBodyMotion``` and ```MultiGrainRigidBodyMotion``` should have a common base class ```GrainRigidBodyMotionBase```. They have a lot in common, like most member variables and parameters.